pkg/clusterdeletion/constraints.go

```diff
@@ -31,6 +31,18 @@ func (d *Deletion) cleanupConstraints(ctx context.Context, cluster *kubermaticv1
 		return err
 	}
 
+	// removing fianlzers from all constraints
+	constraintList := &kubermaticv1.ConstraintList{}
+	if err := d.seedClient.List(ctx, constraintList, ctrlruntimeclient.InNamespace(cluster.Status.NamespaceName)); err != nil {
```

Comment:
Can `cluster.Status.NamespaceName` be empty `""`?

Reply:
every cluster has a namespace in seed so ` cluster.Status.NamespaceName` would not be empty

Reply:
It could be the case that this `Cluster` is deleted right after it was created, or the controller who updates `cluster.Status.NamespaceName` is faulty. In those cases, cluster.Status.NamespaceName can be empty.
so I would suggest to check if this is empty before make API calls, for example, we have some other controllers who are doing similar thing:
https://github.com/kubermatic/kubermatic/blob/master/pkg/controller/seed-controller-manager/addon/addon_controller.go#L134-L136

Reply:
sure will add it

Reply:
done